### DISCUSSION TITLE
some small project ready and in progress

### DISCUSSION THREAD

**Discussion by Jan-Kruis** [postfixadmin/postfixadmin#663] [Ideas]
Hi,

I have bean working on some small project related to PFA.

I'm working on

SendOnly function for mailbox

TLS policy for Domains

To replace special date_strings in the body text with date_from and date_unitl from the vacation table. [in vacation.pl]

To replace friendly name with fullname from the database, if it is set in confile file  [in vacation.pl]

only small symbols no text with help function in template overview for when hovering over the icon's or text(vacation)
	active
	edit
	delete
	sendonly 
	vacation ( no icon only message if vacation is set or not and when set it display de from en until date)

Modifying the template for alias and forwarding to make a differens between de two terms.
         so forwarding uses the display template of users_edit-alias.tpl for both admin and user
        and for the alias funktion the editform.tpl 
( this project is still work in progress)

I add two locatin more for the config.local.php. [etc/mail/postfixadmin/ end /etc/postfixadmin/] this because vacation.pl is looking in this location for vacation.conf this way we can have de two config file inone location. And i am looking for a way to maybe merge the two config file into one this makes the chance of mistakes smaller in my opinion 
( this the merge part is still work in progress)

and a few more minor adjustments
    text translate

Let me know if there is any interest in this adaptation

I now have this change working in my test version. And want to divide them into branches so that it can be better prepared for explanation and comments

so the next step is to see how within github can push branches creation and customization per feature

**Comment by cboltz** [comment]
> Modifying the template for alias and forwarding to make a differens between de two terms.
> so forwarding uses the display template of users_edit-alias.tpl for both admin and user
> and for the alias funktion the editform.tpl

I like the idea, it makes editing mailbox aliases for admins easier.

However my long-term goal is to use `editform.tpl` for (nearly) everything to reduce the number of template files. Of course I can't force you to take this route ;-) but if it doesn't make things too difficult, please consider it.

You'll probably need a condition like `{if $table == 'alias' && $key == 'goto_mailbox'}` with more or less "hardcoded" HTML code for the radio buttons (feel free to copy from `users_edit-alias.tpl`) - but that's still better than having a completely separate template file which also needs to include textarea handling and buttons which you get "for free" with editform.tpl.

BTW: `goto_mailbox` is used in AliasHandler, so you can just hand over the value (`0` or `1`) and AliasHandler will ensure it gets stored correctly.

> I add two locatin more for the config.local.php. [etc/mail/postfixadmin/ end /etc/postfixadmin/] 

I'd only add /etc/postfixadmin/.

(IMHO /etc/mail/postfixadmin/ is deprecated, so no need to add support for another file there.)

> this because vacation.pl is looking in this location for vacation.conf this way we can have de two config file inone location. And i am looking for a way to maybe merge the two config file into one this makes the chance of mistakes smaller in my opinion

I don't think that merging config.local.php and vacation.conf is a good idea - one is PHP (and can even contain PHP function definitions etc.) and the other is perl. I already get headache when I just think about parsing PHP from perl ;-)

**Comment by Jan-Kruis** [comment]
hi christian,

thanks for correcting my typos, i suffer from dyslexia so read about my typos often and swap letters too.

I have already largely finished the adjustment for alias and forward. If you are interested I can email you the adjustment so far

**Reply by cboltz** [reply]
Just open a PR, that's easier than mailing patches around ;-)  (you can keep the PR in draft state until you have everything ready - pushing or force-pushing commits to the branch will update it)

**Comment by Jan-Kruis** [comment]
hi david and christian,

I'm still new to github and don't quite understand how all commands work, 
I'll have to see what you mean by:
(you can keep the PR in draft state until you have everything ready - pushing or force-pushing commits to the branch will update it) 

I had in the code for vacation.pl put a query in the code at line 586 

# QUESTION !! 
# I think the two lines below (which I have commented out in) are no longer necessary, they appear earlier in the script see line 490 and 498 # 
# --> $subject = Encode::encode_utf8($subject) if(Encode::is_utf8($subject)); 
# --> $body = Encode::encode_utf8($body) if(Encode::is_utf8($body));
# 

I think this can go away, if so will you do that or should I make a pull request

Regards 
Jan
